### DISCUSSION TITLE
Debug azure function inside container

### DISCUSSION THREAD

**Discussion by jscheeres** [microsoft/vscode-docker#3429] [Q&A]
I'm trying to debug a azure function written in c# inside a linux container on wsl2. Is this possible? I can't get it working.

When I hit F5 to attach the debugger the extension is copying the debugger into the container.

![image](https://user-images.githubusercontent.com/10026982/154426867-80625868-27c5-487b-a874-95f2cdb13e00.png)

But it throws exceptions.

![image](https://user-images.githubusercontent.com/10026982/154426965-37112427-3a48-4d3e-8cbf-eaffebad5fd7.png)

Dockerfile:

![image](https://user-images.githubusercontent.com/10026982/154427037-8cf4ae9f-dc2a-4048-a4eb-4fa717aa6e80.png)

Launch.json

I added the processName.

![image](https://user-images.githubusercontent.com/10026982/154451000-d79ed9eb-8abe-4f4e-b0aa-48b9a3e60cdf.png)

**Comment by bwateratmsft** [comment]
It is probably possible but might require some manual work to get it going. It looks like the debugger was at least able to attach, which is great progress.

What is the message on those exceptions?

**Comment by jscheeres** [comment]
I can't see the message on those errors. 

The remote debugger has been started.
![image](https://user-images.githubusercontent.com/10026982/154537949-39b943b9-a265-4455-b50f-0012cd3dbb17.png)

When I set a breakpoint, for example on line 20, the debugger doesn't mark the breakpoint in the editor window. And when I do a request to the function it's hitting a breakpoint. But not sure where it's breaking on.

![image](https://user-images.githubusercontent.com/10026982/154538352-ac8a8ea7-fa6e-45ba-a23b-437f5c3083b3.png)

**Reply by bwateratmsft** [reply]
The ghost breakpoint thing is usually do to source maps mismatching. It might work if you change make this change in `launch.json`:

```json
"configurations": [
    {
        "name": "Docker .NET Core Attach (Preview)",
        // ...
        "sourceFileMap": {
            "/src/dotnet-function-app": "${workspaceFolder}" // This is the line that changes, added "/dotnet-function-app"
        }
    },
    // ...
]
```
